uaclient/security.py

```diff
@@ -822,7 +826,7 @@ def _handle_released_package_fixes(
             if not upgrade_status:
                 unfixed_pkgs += [src_pkg for src_pkg, _ in pkg_src_group]
 
-    return upgrade_status, unfixed_pkgs
+    return upgrade_status, unfixed_pkgs, all_already_installed
```

Comment:
I think that's a good approach that you are using. However, we cannot assume that just because we have a binary pkgs to install that they were actually installed.

If you look at `upgrade_packages_and_attach`, if the user is not attached to a ua subscription, we will prompt the user to attach to one to install the package. If the user says no to that, we will not install anything.